### ISSUE TITLE
Disparity results.

### ISSUE THREAD

**Issue by LifeBeyondExpectations** [mileyan/pseudo_lidar#10]
Thanks for sharing this nice work with code.
Can you share the results of the disparity / depth accuracy from your newly modified PSMnet (using KITTI Stereo Benchmark)??
I download your code and achieve the metrics but it is too bad so I am not sure whether the pre-trained weight is correct.
![1](https://user-images.githubusercontent.com/32631453/61346495-fa080800-a893-11e9-8ce7-35d19487d87f.png)

**Comment by mileyan** [comment]
Hi Jaesung,

The released model is trained on 3,712 KITTI 3D object detection training images. And the disparity ground truth is generated by ourselves, which might be slight different with they way used on kitti stereo dataset. You can evaluate it by doing the 3d object detection evaluation.

Best

**Comment by xiazhiyi99** [comment]
> Hi Jaesung,
> 
> The released model is trained on 3,712 KITTI 3D object detection training images. And the disparity ground truth is generated by ourselves, which might be slight different with they way used on kitti stereo dataset. You can evaluate it by doing the 3d object detection evaluation.
> 
> Best

Hello mileyan, thanks for sharing your working! 
I have problems about disparity result too. When I was trying to running your code, I got a terrible result. It turned out to be the PSMnet's problem. So I downloaded your finetuned model, but I got sparse result like this
![image](https://user-images.githubusercontent.com/34741139/67268600-70e21c00-f4e7-11e9-9c1a-1d667347886f.png)
running on 2 1080ti, and here is the command:
```
python ./psmnet/submission.py \
    --loadmodel ./psmnet/kitti_3d/finetune_300.tar \
    --datapath ./KITTI/object/testing/ \
    --save_path ./KITTI/object/testing/predict_disparity
```
It doesnt seem correct, but I really dont know where I got wrong. 
I really appreciate it if you could provide some advise. Forgive my naive.

**Comment by tareeqav** [comment]
I am also seeing the same thing as @xiazhiyi99  with the pretrained PSMNet weights.  My code is:

![disparity_0000000000](https://user-images.githubusercontent.com/53956168/67423862-02fb2a80-f5a3-11e9-9240-6da3c30507f8.png)


```

def infer(model, imgL, imgR, img_idx):
    
    imgL = torch.FloatTensor(imgL).cuda()
    imgR = torch.FloatTensor(imgR).cuda()

    imgL, imgR= Variable(imgL), Variable(imgR)

    imgL, imgR= Variable(imgL), Variable(imgR)

    with torch.no_grad():
        output = model(imgL,imgR)
    output = torch.squeeze(output)
    pred_disp = output.data.cpu().numpy()

    return pred_disp

def main():

    img_idx = 0
    img_idx = str(img_idx).zfill(10)

    processed = preprocess.get_transform(augment=False)

    imgL_o = io.imread(f'./images/left/{img_idx}.png').astype('float32')
    imgR_o = io.imread(f'./images/right/{img_idx}.png').astype('float32')


    imgL = processed(imgL_o).numpy()
    imgR = processed(imgR_o).numpy()
    
    imgL = np.reshape(imgL,[1,3,imgL.shape[1],imgL.shape[2]])
    imgR = np.reshape(imgR,[1,3,imgR.shape[1],imgR.shape[2]])

    # pad to (384, 1248)
    top_pad = 384-imgL.shape[2]
    left_pad = 1248-imgL.shape[3]

    imgL = np.lib.pad(imgL,((0,0),(0,0),(top_pad,0),(0,left_pad)),mode='constant',constant_values=0)
    imgR = np.lib.pad(imgR,((0,0),(0,0),(top_pad,0),(0,left_pad)),mode='constant',constant_values=0)
    
    model = stackhourglass(192)
    model = nn.DataParallel(model, device_ids=[0])
    model.cuda()

    state_dict = torch.load('./weights/psmnet_finetune_300.tar')
    model.load_state_dict(state_dict['state_dict'])

    print('Number of model parameters: {}'.format(sum([p.data.nelement() for p in model.parameters()])))

    model.eval()
    pred_disp = infer(model, imgL, imgR, img_idx)
     
    top_pad   = 384-imgL_o.shape[0]
    left_pad  = 1248-imgL_o.shape[1]
    img = pred_disp[top_pad:,:-left_pad]
    io.imsave(f'images/disparity_{img_idx}.png',(img*256).astype('uint16'))
    
if __name__ == '__main__':
    main()
```

**Comment by sarimmehdi** [comment]
Hello. I am getting same results as above with the pre-trained model. The depth image looks like a mess. I am running it on new images however but the results are still quite bad. Is this normal?
